projects/storefrontlib/src/cms-components/cart/add-to-cart/add-to-cart.component.ts

```diff
@@ -74,8 +75,18 @@ export class AddToCartComponent implements OnInit {
     if (!this.productCode || this.quantity <= 0) {
       return;
     }
-    this.openModal();
-    this.cartService.addEntry(this.productCode, this.quantity);
+    // check item is already present in the cart
+    // so modal will have proper header text displayed
+    this.cartService
+      .getEntry(this.productCode)
+      .subscribe(entry => {
+        if (entry) {
+          this.increment = true;
```

Comment:
Need to set this to false at the end of the subscribe or you'll never get the old "item(s) added to your cart" again.

Reply:
right, great spot here! already pushed a change